src/doom/d_main.c

```diff
@@ -1706,6 +1704,12 @@ void D_DoomMain (void)
     if (M_ParmExists("-dehlump"))
     {
         int i, loaded = 0;
+        int numiwadlumps = numlumps;
+
+        while (!W_IsIWADLump(lumpinfo[numiwadlumps - 1]))
+        {
+            numiwadlumps--;
```

Comment:
What prevents this going negative?

Reply:
that we have an IWAD with at least one lump loaded at this point